src/frontend/js/modules/sidebar.js

```diff
@@ -30,8 +35,13 @@
       return contextualSidebarService;
     })
 
-    .directive('contextualSidebar', function($timeout, $window, $mdUtil, contextualSidebarService, session) {
+    .directive('contextualSidebar', function($timeout, $window, $mdUtil, contextualSidebarService, session, searchProviders) {
       function link(scope, element, attr) {
+        // Check if we need to show the global search box
+        if (searchProviders.providersPromises.length > 0) {
+          scope.showGlobalSearch = true;
+        }
```

Comment:
```suggestion
    scope.showGlobalSearch = searchProviders.providersPromises.length > 0;
```
Nit: And the same suggestion by having a dedicated method to get list of providers